### ISSUE TITLE
Wallet sync issues

### ISSUE THREAD

**Issue by shikita** [spesmilo/electrum#5489]
Thank you.

Using latest electrum and X.

Wallet having around 20k addresses  and its unable to sync.  While synching up to around 5k then it again syncs from starting number.  it resets again  to **"Synchronizing... (343/20100)"**.  What might be the issue.

**Comment by SomberNight** [comment]
Did it work before?

Check the client [logs](https://github.com/spesmilo/electrum-docs/blob/master/faq.rst#how-to-enable-debug-logging).

> Using latest electrum and X.

Do you run the server yourself? ElectrumX?
By default, there are [resource limits for sessions](https://electrumx.readthedocs.io/en/latest/environment.html#envvar-COST_SOFT_LIMIT).
You can disable them:
```
COST_SOFT_LIMIT=0
COST_HARD_LIMIT=0
```

**Comment by shikita** [comment]
I enabled the log and retried it.

`201907 |     INFO | interface.[shikita_rasp:50001] | disconnecting due to RequestTimedOut("request timed out: ('blockchain.scripthash.subscribe', ['redacted_script_hash']) (id: 6189)")`

**Comment by shikita** [comment]
i tried ` REQUEST_TIMEOUT=9999`  but no use.

**Comment by SomberNight** [comment]
Is your server public or private? (so e.g. can it be found via server peer discovery?)
Are you connecting to it on a LAN, or even localhost? Or e.g. over internet, via Tor / VPN / etc?
Did you try disabling the resource limits like I said?

Can you sync a smaller wallet successfully?

It would be useful if you were less terse.

**Comment by shikita** [comment]
The server is private. Its on my localhost on raspberrypi.
Yes smaller wallets sync was successfull.
Config file of electrumx.

```
export COIN=BitcoinSegwit
export PEER_DISCOVERY=off
export PEER_ANNOUNCE=''
export DB_ENGINE=leveldb
export DB_DIRECTORY=/media/shikita/sonyssd/xdb/
export ALLOW_ROOT=allow

export DAEMON_URL=http://shikita:shikita123@localhost:8332

export SERVICES=tcp://0.0.0.0:50001,rpc://localhost:8000
export COST_SOFT_LIMIT=0
export COST_HARD_LIMIT=0
export REQUEST_TIMEOUT=9999
```

**Comment by SomberNight** [comment]
Try starting the client with `--oneserver`

**Comment by shikita** [comment]
Tried `--oneserver`. Got the same error `RequestTimedOut`

**Comment by shikita** [comment]
Electrumx log: https://paste.ubuntu.com/p/5jnmd7xSQj/

**Comment by shikita** [comment]
Is there a request time out set on client side?

**Comment by shikita** [comment]
While starting electrum wallet i got the following warning. Is it related to this issue?

```
/usr/local/lib/python3.7/site-packages/google/protobuf/descriptor.py:47: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from google.protobuf.pyext import _message
```

**Comment by shikita** [comment]
Update:  retried again to sync and it synced all 20k addresses but after 20k then again  it resets to 0 and resyncs :(

**Comment by fenidik** [comment]
Same for me. Sometimes Electrum stuck and writing in log

```20201014T165520.952476Z |     INFO | interface.[somehost:50001] | disconnecting due to RequestTimedOut("request timed out: ('blockchain.scripthash.subscribe', ['somehash']) (id: 12)")```

it lasts until I remove all Electrum data and sync wallet again from scratch.

After few days it happens again.

Tried with ElectrumX and Electrs - same result.

Electrum 3.3.8 
wallet with 100k addresses

**Comment by ln2max** [comment]
This happens to me as well, especially if the connection to the server is not very good (using a server behind a Tor onion service for example).

2 ideas for solving the issue:
- make it possible to set the client-side timeout value (for bad connections or Tor services, this value should be very large)
- make it possible to resume synchronization: if the connection drops after syncing addresses 1 to 50000, then when reconnecting start synchronizing at address 50001. Don't start over at the beginning until the wallet has fully synced.

I'll make a separate issue for the 2nd one, as it's a significant feature.
